%% START INTRO CHAPTER
\addcontentsline{toc}{chapter}{\introtitle}
\chapter*{\introtitle}
\clearpage

Although they are typically associated with the study of the evolution of species, phylogenetic trees can be used to study the evolution of any sequence that evolves. For example, phylogenetic methods have been used to study the evolution of multicopy gene families~ \cite{Page1997}, cancer genomes~\cite{El-Kebir2016,Nowell1976}, antibodies~\cite{Litman1993,Robinson2015,Safonova2015}, segmental duplicates~\cite{Bailey2006,Jiang2007}, and transposable genomic elements~\cite{Dewannieux2003,Moshiri2017}, which are all entities that evolve \textit{within} the genome of a single species. Further, they can be used to study the evolution of viruses, both within and across hosts~\cite{Frost2001,Lemey2006,Vrancken2014}. In the case of viruses, the history of transmission events constrains the evolutionary history, such as imposing a bottleneck at the time of each transmission~\cite{Carlson2014}.

Models of tree evolution describe probability distributions over the space of tree shapes~\cite{Yule1925,Aldous2001}, which can be used as the prior distribution in a Bayesian inference~\cite{Drummond2007,Mooers2012,Sayyari2016}, to generate null distributions describing certain neutral processes~\cite{Guyer1991,Kirkpatrick1993,Agapow2002}, or to infer evolutionary parameters inherently of interest to the biologist~\cite{Morlon2014}. Similarly, generalized epidemic models describe probability distributions over the space of transmission networks~\cite{Sahneh2013}, and simulations that sample the distributions defined by these stochastic models allows epidemiologists to study infection patterns of disease epidemics~\cite{Sahneh2017}. Further, network models describe probability distributions over graphs, and they can be used to capture features of networks of interactions (e.g. social interactions between humans)~\cite{Watts1998,Watts1999,Barabasi1999,Erdos1959}. Lastly, models of sequence evolution describe the mutation of a sequence over time~\cite{Jukes1969,Kimura1980,Felsenstein1981,Tamura1993,Tavare1986}, and they can be used to simulate the evolution of a sequence down a phylogeny~\cite{Rambaut1997} as well as to infer the evolutionary history of a set of sequences~\cite{Felsenstein2003}.

The spread of many infectious diseases is driven by social and sexual networks~\cite{Rivas2012}, and reconstruction of their transmission histories from molecular data can greatly enhance intervention. For example, network-based statistics for measuring \gls{HIV} treatment effects can yield increased statistical power~\cite{Wertheim2011}; the analysis of the growth of \gls{HIV} infection clusters can yield actionable epidemiological information for disease treatment and prevention~\cite{Aldous2012,Brenner2013}; transmission-aware models can be used to infer rates of \gls{HIV} evolution~\cite{Vrancken2014}. The ability to infer properties and patterns of the transmission history of a viral epidemic allows public health officials to intervene and attempt to prevent the spread of the virus. In the case of \gls{HIV}, patients who adhere to \gls{ART} can become ``virally suppressed,'' meaning the virus is kept at bay, resulting in slower progression of the \gls{HIV} disease as well as a significant reduction in transmission risk~\cite{AIDSinfo2019}. Thus, the ability to predict which individuals are most at-risk of transmitting the virus would provide public health officials actionable information: they can take measures to ensure high-risk individuals are able to continuously adhere to \gls{ART}.

The ability to infer and reconstruct properties of a transmission network has been researched extensively in recent years~\cite{Wertheim2017,Ragonnet-Cronin2019}, and many tools exist that attempt to use molecular data to try to perform this inference~\cite{Rose2017}. For example, PhyloPart~\cite{Prosperi2011}, Cluster Picker~\cite{Ragonnet-Cronin2013}, and TreeCluster~\cite{Balaban2019} infer transmission clusters using phylogenies inferred from viral sequences. HIV-TRACE, on the other hand, infers transmission clusters directly from sequences~\cite{Pond2018}. While these tools have been used to analyze real datasets~\cite{Campbell2017}, the accuracies, errors, and limitations of these methods are still poorly understood.

By utilizing models of social contact networks, viral transmission, tree evolution, and sequence evolution together, epidemiologists can define complex probabilistic distributions composed of sub-models. These complex distributions can be sampled to simulate realistic data representative of a virus of interest as it spreads through a population of interest, and the resulting data can be used to evaluate the accuracies of transmission network inference methods as well as to study trends and patterns of an epidemic as a function of the various model parameters to gain insights into the mechanisms driving the epidemic of interest~\cite{Ratmann2017}. However, many existing tools to perform such epidemic simulations have author-imposed model assumptions that the user cannot relax or change. In Chapter~\ref{chap:favites}, I will discuss FAVITES: a novel epidemic simulation framework I developed that makes absolutely no model assumptions about the epidemic. The framework is defined by a series of interactions of abstract modules, and each \textit{implementation} of a module defines the model assumptions. Thus, users are free to select whichever module implementations (and thus model assumptions) that best fit their epidemic of interest. In addition to presenting the tool, I will describe in detail a simulation experiment designed to emulate the San Diego \gls{HIV} epidemic between 2005 and 2014, and I will use the simulated data to compare and contrast existing transmission clustering methods.

Of course, the ability to simulate an epidemic depends entirely on the existence of statistical models that appropriately describe the processes of the epidemic of interest. In the case of retroviruses and retrotransposons, which replicate via reverse transcription~\cite{Whitcomb1992} and may undergo significant selection pressure~\cite{Wood2009}, a neutral model of tree evolution like the Yule~\cite{Yule1925} or Coalescent~\cite{Kingman1982} may not be appropriate. In Chapter~\ref{chap:dualbirth}, I will discuss the dual-birth model, a novel model of tree evolution that is motivated by the retrotransposition of \textit{Alu} elements in the human genome~\cite{Batzer2002}. I will derive various probabilistic distributions and theoretical expectations of trees sampled under the model, and I will then present two approaches for estimating model parameters from a given phylogeny. I will then present the results of an analysis of close to one million \textit{Alu} sequences from the human genome in which I infer the dual-birth model parameters and present an estimate of the number of active \textit{Alu} elements, a topic of much debate~\cite{Price2004,Deininger2011,Cordaux2004}.

The goal of many transmission clustering analyses is to learn about the dynamics of a virus through a given population, often to try to predict which sub-populations may be spreading the virus more rapidly~\cite{Wertheim2011,Wertheim2017,Little2014}. However, transmission clustering is essentially a way of summarizing the relationships between the sampled viral sequences, and instead of performing predictions and inferences on these summaries, what if we were able to infer properties of interest directly from the evolutionary relationships of the viruses? In Chapter~\ref{chap:proact}, I investigate a single specific question: Given a set of viral sequences sampled from people living with \gls{HIV}, can I predict which individuals are most at-risk of transmitting the virus in the future? In an attempt to address this question, I present ProACT, a tool that attempts to prioritize people living with \gls{HIV} based on risk of future transmissions. ProACT depends only on the viral phylogeny and does not require any demographic information from the patients, meaning it is not sensitive to error-prone survey data, and most importantly, it is less prone to bias.

A primary focus of mine is the ability to perform such analyses in a massively-scalable fashion. The scalability of a computational tool is primarily dependant on two things: (1) the theoretical time complexity of the algorithm, and (2) the efficiency of the implementation of the algorithm. While developing FAVITES and ProACT, I found that, although the phylogenetic algorithms I designed were quite fast in theory, my implementations using existing tree-manipulation packages were much slower than I anticipated due to significant overhead in loading and initializing my ultra-large phylogenies. In Chapter~\ref{chap:treeswift}, I will present TreeSwift, my own Python package for traversing and manipulating trees. I will describe its implementation design, demonstrate some of its features, and compare its execution time for various common tree algorithms against existing packages.

As can be seen, as the cost of sequencing decreases, the amount of viral sequence data available to researchers is growing rapidly, and as a result, the field of Epidemiology is becoming increasingly dependent on scalable computational methods. However, many researchers in the fields of Epidemiology and Molecular Biology have never received formal education in computation. In recent years, computational courses have started appearing in undergraduate Biology major curricula, and while this will provide computational skills to the next generation of biomedical and epidemiological researchers, it does not benefit the current generation of professionals. In Chapter~\ref{chap:education}, I will present my contributions to Bioinformatics education in the form of developing novel \glspl{MOOC} and \glspl{MAIT}, and I will discuss the pedagogical philosophy I employed in developing my learning materials.

In summary, I show that modern studies in viral epidemiology require the ability to perform simulation experiments that can appropriately capture the virus and population of interest, which thus requires tools to run such simulations efficiently as well as statistical models that make realistic assumptions. I also show that the ability to infer actionable epidemiological information from molecular data is an open problem. In this dissertation, I present a novel framework for epidemic simulations, provide a novel model of phylogenetic evolution (and derive probabilistic distributions and theoretical expectations of trees sampled under the model), demonstrate the effectiveness of a novel phylogenetic tool for prioritizing people living with \gls{HIV} based on their risk of future transmissions, and introduce a novel package for performing tree traversals and manipulations efficiently on ultra-large phylogenies. I also discuss my contributions to Bioinformatics education.

%% END INTRO CHAPTER