%% Dissertation Info
\title{Mathematical Modeling of Viral Evolution and Epidemiology}
\author{Alexander Niema Moshiri}
\degreeyear{2019}
\degreetitle{Doctor of Philosophy}
\field{Bioinformatics and Systems Biology}
%\specialization{Anthropogeny}  % If you have a specialization, add it here

%% Committee Info
\chair{Professor Siavash Mirarab}
\cochair{Professor Pavel A. Pevzner}
\othermembers{
Professor Vineet Bafna\\
Professor David M. Smith\\
Professor Joel O. Wertheim\\
}
\numberofmembers{5} % |chair| + |cochair| + |othermembers|

%% Define Acknowledgement Statements (to avoid typos when copying)
\newcommand{\ackfavites}{Chapter~\ref{chap:favites}, in full, is a reprint of the material as it appears in ``\favitestitle'' (2018). Moshiri, Niema; Ragonnet-Cronin, Manon; Wertheim, Joel; Mirarab, Siavash, \textit{Bioinformatics}, bty921. The dissertation author was the primary investigator and first author of this paper.}
\newcommand{\ackdualbirth}{Chapter~\ref{chap:dualbirth}, in full, is a reprint of the material as it appears in ``\dualbirthtitle'' (2017). Moshiri, Niema; Mirarab, Siavash, \textit{Systematic Biology}, 67(3), 475-489. The dissertation author was the primary investigator and first author of this paper.}
\newcommand{\ackproact}{Chapter~\ref{chap:proact}, in full, has been submitted for publication of the material as it may appear in ``\proacttitle'' (2019) Moshiri, Niema; Smith, Davey; Mirarab, Siavash, \textit{Molecular Biology and Evolution}. The dissertation author was the primary investigator and first author of this paper.}
\newcommand{\acktreeswift}{Chapter~\ref{chap:treeswift}, in full, has been submitted for publication of the material as it may appear in ``\treeswifttitle'' (2019) Moshiri, Niema, \textit{SoftwareX}. The dissertation author was the primary investigator and sole author of this paper.}

%% START FRONTMATTER
\begin{frontmatter}
\makefrontmatter

%% Dedication
\begin{dedication}
\vspace*{\fill}
I dedicate this dissertation to Ladan Moshiri, Ramin Moshiri, and Michelle Roxanna Moshiri-Warncke for years of love and support.
~\\~\\~\\
I dedicate this dissertation to Karen George for always being by my side.
~\\~\\~\\
I dedicate this dissertation to Ryan Micallef and Felix Garcia for teaching me to live my life a quarter mile at a time.
~\\~\\~\\
I dedicate this dissertation to all my friends in the Bioinformatics and Systems Biology program for some of the best memories of my life.
\vspace*{\fill}
\end{dedication}

%% Epigraph
\begin{epigraph}
\begin{center}
\begin{minipage}{0.65\linewidth}
\onehalfspacing
{\large
\textit{Folk in these stories had lots of chances of turning back, only they didn't. Because they were holding onto something.}\\
~\\
\textit{What are we holding onto, Sam?}\\
~\\
\textit{That there's some good in this world, Mr. Frodo. And it's worth fighting for.}
\begin{flushright}
---J.R.R. Tolkien
\end{flushright}
}
\end{minipage}
\end{center}
\end{epigraph}

%% Table of Contents
\tableofcontents

%% List of Abbreviations
\newpage
\addcontentsline{toc}{chapter}{\abbrevtitle}
\begin{center}\expandafter\MakeUppercase\expandafter{\abbrevtitle}\end{center}
\let\clearpage\relax
\vspace*{-2cm}
\printglossary[type=\acronymtype,title={}]

%% List of Symbols

%% List of Figures and Tables
\listoffigures
\listoftables


%% Acknowledgements
\begin{acknowledgements}
I would like to thank all of the people who have helped me throughout my Ph.D. My colleagues in the Bioinformatics and Systems Biology Graduate Program and in the Computer Science and Engineering department have been great to work with, especially Jens Luebeck, Margaret Donovan, Bill Greenwald, Ben Pullman, and Argus Athanas. I enjoyed having the support of such excellent lab mates as Erfan Sayyari, Metin Balaban, Uyen Mai, Maryam Rabiee, Sina Malekian, and Chao Zhang.
I would also like to thank Prof.~Joel Wertheim, Manon Ragonnet-Cronin, Prof.~Davey Smith, and Prof.~Sanjay Mehta for teaching so much about the molecular biology and epidemiology of HIV.
I am truly blessed to have been able to work with such amazing people.

I would not be where I am today (both figuratively and literally) without the mentorship of Prof.~Vineet Bafna. I met him early on in my undergraduate career, and he has been the source of sage wisdom throughout my entire academic career thus far. It was largely because of him that I decided to continue to pursue graduate education in Bioinformatics, and I am thankful that his guidance led me down this path.

I am grateful of Prof.~Pavel Pevzner for taking me under his wing at a time when I was somewhat lost regarding my career aspirations. Growing up, I always thought I wanted to become a medical doctor, but I always had a strong passion for technology. It was under his mentorship that I recognized the possibility to merge these two seemingly unrelated fields, and I was able to see the biomedical applications of Computer Science. He also introduced me to the field of Bioinformatics Education, and it was because of his guidance that I realized my passion for teaching, and I am honored he took a chance on me. He is truly a visionary, and I am happy to have been able to work with him.

When starting a Ph.D. program, one of the most stressful thoughts is ``Will I find a good mentor?'', and I was extremely fortunate to have found Prof.~Siavash Mirarab. I cannot thank him enough for all the time and effort he has taken to make my Ph.D. experience productive and intellectually stimulating. From the long sessions in which we'd be working on an algorithmic problem together, to the hours he would spend helping me revise my manuscripts, the effort he took to mentor me has instilled within me far more than I could have imagined. I aspire to be as brilliant of a scientist as he one day, and I hope to be as great of a mentor as he was to me.

\ackfavites

\ackdualbirth

\ackproact

\acktreeswift
\end{acknowledgements}

%% Vita
\begin{vitapage}
\begin{vita}
\item[2015] B.~S. in Bioengineering: Bioinformatics, University of California, San Diego
\item[2015-2016] Teaching Assistant, University of California, San Diego
\item[2017] Associate Instructor, University of California, San Diego
\item[2015-2019] Research Assistant, University of California, San Diego
\item[2019] Ph.~D. in Bioinformatics and Systems Biology, University of California, San Diego
\end{vita}
\begin{publications}
\item \textbf{Niema Moshiri} (2019). ``Why Do I Look Like My Mom and Dad? Plants and Animals Inherit Traits From Parents,'' \textit{STEMTaught}
\item Metin Balaban, \textbf{Niema Moshiri}, Uyen Mai, and Siavash Mirarab (2019). ``TreeCluster: Clustering Biological Sequences using Phylogenetic Trees,'' \textit{GLBIO 2019}, doi:~10.1101/591388
\item Adam Rule, Amanda Birmingham, Cristal Zuniga, Ilkay Altintas, Shih-Cheng Huang, Rob Knight, \textbf{Niema Moshiri}, Mai H. Nguyen, Sara Brin Rosenthal, Fernando PÃ©rez, and Peter W. Rose (2019). ``Ten Simple Rules for Reproducible Research in Jupyter Notebooks,'' \textit{PLOS Computational Biology (in press)}, arXiv:~1810.08055
\item \textbf{Niema Moshiri} and Liz Izhikevich (2018). \textit{Design and Analysis of Data Structures}, Amazon KDP, ISBN:~1981017232
\item \textbf{Niema Moshiri}, Manon Ragonnet-Cronin, Joel O. Wertheim, and Siavash Mirarab (2018). ``FAVITES: simultaneous simulation of transmission networks, phylogenetic trees, and sequences,'' \textit{Bioinformatics}, doi:~10.1093/bioinformatics/bty921
%\item \textbf{Niema Moshiri} (2018). ``The dual-Barab\'asi-Albert model,'' \textit{arXiv}, arXiv:~1810.1810.10538
%\item \textbf{Niema Moshiri} (2018). ``TreeN93: a non-parametric distance-based method for inferring viral transmission clusters,'' \textit{bioRxiv}, doi:~10.1101/383190
\item \textbf{Niema Moshiri} (2018). ``TreeSwift: a massively scalable Python tree package,'' \textit{bioRxiv}, doi:~10.1101/325522
\item \textbf{Niema Moshiri}, Liz Izhikevich, and Christine Alvarado (2018). ``Data Structures: An Active Learning Approach,'' \textit{edX}
\item \textbf{Niema Moshiri}, Phillip Compeau, and Pavel Pevzner (2017). ``Analyze Your Genome!,'' \textit{edX}
\item Phillip Compeau, \textbf{Niema Moshiri}, and Pavel Pevzner (2017). ``Introduction to Genomic Data Science,'' \textit{edX}
%\item \textbf{Niema Moshiri} (2017). ``A linear-time algorithm to sample the dual-birth model,'' \textit{bioRxiv}, doi:~10.1101/226423
\item \textbf{Niema Moshiri} and Siavash Mirarab (2017). ``A Two-State Model of Tree Evolution and Its Applications to Alu Retrotransposition,'' \textit{Systematic Biology}, 67(3), 475-489. doi:~10.1093/sysbio/syx088
\end{publications}
\end{vitapage}

%% Abstract
\begin{abstract}
Phylogenetic trees can be used to study the evolution of any sequence that evolves, including viruses. In a viral epidemic, the history of transmission events defines constraints on the evolutionary history of the viral population. Models of tree evolution describe probability distributions over the space of tree shapes and can be used to simulate trees that capture the evolutionary patterns of real world phenomena as well as to infer evolutionary parameters inherently of interest to the biologist. The spread of many viruses is driven by social and sexual networks, and because of the relationship between their evolutionary and transmission histories, phylogenetic inference from viral sequences can be used to improve the inference of patterns of the epidemic, which in turn can greatly enhance epidemiological intervention. The simultaneous simulation of viral transmission network, phylogenetic tree, and sequences can provide a method to observe the effects of virus model parameters on the epidemic as well as to study the accuracies and errors of transmission inference tools. Further, the development of massively-scalable tools to analyze ultra-large datasets of viral sequences can aid epidemiologists in the real-time surveillance of the spread of disease. In an effort to contribute to these ideas, I developed a novel framework to simulate viral transmission networks, phylogenetic trees, and sequences, developed a novel evolutionary model, studied the effects of model parameters on epidemic outcomes via simulation experiments using the proposed framework, developed a scalable and non-parametric method of transmission risk prioritization, and contributed to publicly-accessible Bioinformatics education.
\end{abstract}
\end{frontmatter}

%% END FRONT MATTER