%% START TREESWIFT CHAPTER
\chapter{\treeswifttitle}
\label{chap:treeswift}
\clearpage

Phylogenetic trees are essential to evolutionary biology, and numerous methods exist that attempt to extract phylogenetic information applicable to a wide range of disciplines, such as epidemiology and metagenomics. Currently, the three main Python packages for trees are Bio.Phylo, DendroPy, and the ETE Toolkit, but as dataset sizes grow, parsing and manipulating ultra-large trees becomes impractical for these tools. To address this issue, I developed TreeSwift, a user-friendly and massively scalable Python package for traversing and manipulating trees that is ideal for algorithms performed on ultra-large trees.

\section{Motivation and Significance}\label{sec:treeswift-background}
Phylogenetic trees are essential to evolutionary biology, and phylogenetic methods are applicable to a wide range of disciplines, such as epidemiology \cite{Ragonnet-Cronin2013,Rose2017} and metagenomics \cite{Kembel2011,Darling2014,Filipski2015}. However, the datasets analyzed by these methods are growing rapidly as sequencing costs continue to fall, emphasizing the need for scalable methods of tree traversal and manipulation. Beyond the analysis of real datasets, phylogenetic approaches can be utilized in the analysis of potentially massive datasets generated by simulation experiments~\cite{Moshiri2018}.

Methods for performing phylogenetic analyses such as clustering~\cite{Balaban2019} and rerooting~\cite{Mai2017} are typically presented as a series of higher-level tree traversals and manipulations. The developers of these tools do not commonly implement basic tree processing from scratch: they typically utilize existing tree packages to handle low-level tasks and instead implement their algorithms as a series of calls to functions of these packages. As a result, the performance of such a tool depends not only on the time complexity of its algorithm, but also on the performance of the underlying tree package.

Currently, the three main Python packages for trees are the Bio.Phylo module of Biopython~\cite{Cock2009}, DendroPy~\cite{Sukumaran2010}, and the ETE Toolkit~\cite{Huerta-Cepas2016}. The three tools are simple to integrate into new methods, include a plethora of functions that cater to most phylogenetics needs, and are fast for reasonably-sized trees. However, as dataset sizes grow, parsing and manipulating ultra-large trees becomes impractical. I introduce TreeSwift, a scalable cross-platform Python package for traversing and manipulating trees that does not require any external dependencies, and I compare its performance against that of Bio.Phylo, DendroPy, and the ETE Toolkit.

\section{Software Description}\label{sec:treeswift-description}
\subsection{Software Overview}\label{sec:treeswift-overview}
TreeSwift is a pure-Python package that has no required external dependencies and which has been tested on Python versions 2.6--2.7 and 3.3--3.7. It is also compiled and hosted on PyPI, meaning it can easily be installed with a single \texttt{pip} command without any need for administrative privileges or any advanced knowledge. This is essential to contrast against the current state-of-the-art, ETE Toolkit, which requires the Six and NumPy Python libraries to install if the user has administrative privileges or Anaconda/Miniconda to install if the user doesn't, and BioPython, which requires a C compiler and the NumPy Python library as well as the computer fluency to compile tools from source using a \texttt{Makefile}.

A key feature of TreeSwift is its simplicity in class design in order to reduce time and memory overhead of loading, traversing, and manipulating trees. The entire package consists of just two classes: a \texttt{Node} class, which contains the data and local relationships, and a \texttt{Tree} class, which handles manipulation and traversal on the \texttt{Node} objects. A key distinction between TreeSwift and DendroPy is that DendroPy stores bipartition information to enable efficient comparisons between multiple trees that share the same set of taxa, but because TreeSwift is designed for the fast traversal and manipulation of individual trees (and not for the comparison of multiple trees), TreeSwift forgoes this feature to avoid the accompanied overhead, resulting in a much lower memory footprint and faster execution of equivalent functions (Fig.~\ref{fig:treeswift-comparison}).

\begin{figure} % FIGURE 1 IN ORIGINAL PAPER
\centering
\includegraphics[width=0.7\textwidth]{figs/treeswift-comparison}
\caption[Runtime Comparison]
{Runtimes of DendroPy, Bio.Phylo, the ETE Toolkit, and TreeSwift for a wide range of typical tree operations using trees of various sizes, as well as memory consumption after loading a tree. The truncation of a given tool's plot implies lack of scalability beyond that point, and the entire lack of a given tool implies lack of implementation of the tested functionality. Timing was performed on a computer running CentOS release 6.6 (Final) with an Intel(R) Xeon(R) CPU E5-2670 0 at 2.60GHz and 32 GB of RAM.}
\label{fig:treeswift-comparison}
\end{figure}

\subsection{Software Functionalities}\label{sec:treeswift-functions}
TreeSwift supports loading trees in the Newick, Nexus, and NeXML file formats via the \texttt{read\_tree\_newick}, \texttt{read\_tree\_nexus}, and \texttt{read\_tree\_nexml} functions, respectively. Inputs to these functions can be strings, plaintext files, or gzipped files, and TreeSwift handles the nuances of parsing them internally to maintain user-friendly operability.

TreeSwift provides generators that iterate over the nodes of a given tree in a variety of traversals, including pre-order, in-order, post-order, level-order, and root-distance-order. TreeSwift also allows for the modification of the structure of a given tree by simply modifying the \texttt{Node} objects of the tree. These built-in generators and modifiers intend to provide developers a simple yet efficient manner in which to implement their own algorithms such that they only need to consider higher-level details of the traversal process.

TreeSwift also provides the ability to compute various summarizing statistics of a given tree, such as tree height, average branch length, patristic distances between nodes in the tree, treeness~\cite{Phillips2003}, and the Gamma statistic~\cite{Pybus2000}. Beyond numerical statistics to describe trees, TreeSwift can also generate a visual summary of a tree in the form of a \gls{LTT} plot~\cite{Harvey1994}, a feature not currently implemented in any other Python tree package.

\section{Illustrative Example}\label{sec:treeswift-example}
In the following example, I load a tree from a gzipped file, compute the minimum distance from each node in the tree to a leaf, print the minimum leaf distance of the root, and create a \gls{LTT} plot (Fig.~\ref{fig:treeswift-ltt}).
\clearpage

\begin{python}
from treeswift import read_tree_newick
tree = read_tree_newick("my_huge_tree.nwk.gz")
min_leafdist = dict()
for u in tree.traverse_postorder():
    if u.is_leaf():
        min_leafdist[u] = 0
    else:
        min_leafdist[u] = min(min_leafdist[c]+c.edge_length for c in u.children)
print("Minimum leaf distance from root: %f" % min_leafdist[tree.root])
tree.lineages_through_time(color="blue")
\end{python}

\begin{figure} % FIGURE 2 IN ORIGINAL PAPER
\centering
\includegraphics[width=0.7\textwidth]{figs/treeswift-ltt}
\caption[Lineages Through Time]
{Example \gls{LTT} plot generated using TreeSwift.}
\label{fig:treeswift-ltt}
\end{figure}

\section{Impact}\label{sec:treeswift-impact}
The key impact of TreeSwift is its significant performance improvement over existing Python tree packages (Fig.~\ref{fig:treeswift-comparison}). For almost all tested tree operations, TreeSwift performed tasks significantly faster than all existing tools (by orders of magnitude at times), and it was the only tool that not only had all tested functions implemented, but that also was able to scale to the largest of tested datasets. Further, TreeSwift's memory consumption was significantly lower than all existing tools. Thus, phylogenetic tools written in Python can utilize TreeSwift for scalability.

Further, TreeSwift was designed to be simple to use. As can be seen in the example code in Section~\ref{sec:treeswift-example}, a user with minimal Python experience can generate a \gls{LTT} plot in just 3 lines of Python code. Even complex tree algorithms can be implemented cleanly by utilizing TreeSwift's traversal generators \cite{Balaban2019}.

It must be emphasized that, although TreeSwift was designed with the field of phylogenetics in mind, the package is general in that it can be utilized with any arbitrary tree structure, including those in non-phylogenetic applications~\cite{Moshiri2018b}. Thus, its utility can extend well beyond its intended phylogenetics audience.

\section{Conclusions}\label{sec:treeswift-conclusions}
In this article, I presented TreeSwift, a pure-Python package for loading, traversing, and manipulating trees in a massively-scalable manner. The current version implements a wide range of typical tree operations, and due to its simple design, I hope to engage other developers to further expand TreeSwift's capabilities to target a larger suite of potential applications.

\section{Acknowledgements}
This work was supported by NIH subaward 5P30AI027767-28 to NM. I would like to acknowledge Siavash Mirarab for his mentorship. I would also like to acknowledge Jeet Sukumaran and Mark Holder, as DendroPy provided much motivation during TreeSwift's development.

\acktreeswift

%% END TREESWIFT CHAPTER